pkg/common/configs/configs.go

```diff
@@ -53,3 +52,51 @@ func (ctx *SchedulerConfigContext) Get(policyGroup string) *SchedulerConfig {
 	defer ctx.lock.RUnlock()
 	return ctx.configs[policyGroup]
 }
+
+func AddConfigMapCallback(id string, callback func()) {
+	configMapLock.Lock()
+	defer configMapLock.Unlock()
+	configMapCallbacks[id] = callback
+}
+
+func RemoveConfigMapCallback(id string) {
+	configMapLock.Lock()
+	defer configMapLock.Unlock()
+	delete(configMapCallbacks, id)
+}
```

Comment:
2 things:
* Comments are missing
* When are these functions called? I can see only one reference from the unit test.

Reply:
I'll add the comments.

Currently, nothing calls these, as we don't have any config vars specific to the core. See comments on https://issues.apache.org/jira/browse/YUNIKORN-1213 for the first likely candidate (controlling health check interval).

The idea is that code which can respond to config changes will get a callback when new values are loaded, and can update internal state.